New LDAP attributes in accordance with Scolaris

paraenggu opened the pull request:
The following PR adds additional LDAP attributes in accordance with Scolaris and customer requirements.

The following attributes were added:
* `base4kidsGender`: Gender of a person
* `base4kidsPersiskaNumber`: Employees reference number from the information system PERSISKA
* `base4kidsExpirationDate`: Expiration date of a Base4Kids object in coordinated universal time
* `base4kidsSchoolClassYear`: The school class year
* `base4kidsSchoolProgrammYear`: The school programm year (Programmjahr)
* `base4kidsLegalGuardianOf`: References to DNs for which a person is a legal guardian
* `base4kidsLegalGuardian`: References to legal guardian DNs of a person
* `base4kidsSchoolLevel`: Represents the level of a class
* `base4kidsSchoolCycle`: Defines to which cycle a given school class belongs to

paraenggu committed:
Adding base4kidsGender attribute

paraenggu committed:
Adding base4kidsPersiskaNumber attribute

The base4kidsPersiskaNumber is an employees reference number originating from
the PERSISKA information system (Personalinformationssystems PERSISKA).

paraenggu committed:
Adding base4kidsExpirationDate attribute

paraenggu committed:
Adding base4kidsSchoolClassYear attribute

paraenggu committed:
Updated copyright header to reflect the current year.

paraenggu committed:
Adding base4kidsSchoolProgrammYear attribute

The base4kidsSchoolProgrammYear attribute represents a pupil's current school
programm year (Programmjahr), from 1 to 9.

paraenggu committed:
Adding base4kidsLegalGuardianOf attribute

The base4kidsLegalGuardianOf holds one or more references to pupils for which
this person has a legal guardian relationship.

paraenggu committed:
Adding base4kidsLegalGuardian attribute

The base4kidsLegalGuardian holds one or more references to person entries
acting as a legal guardian for this entry.

paraenggu committed:
Adding base4kidsSchoolLevel attribute

The base4kidsSchoolLevel attribute represents the level of a given class, such
as KINDERGARTEN, BASISSTUFE, PRIMARSTUFE, SEKUNDARSTUFE1 and
SONDERSCHULKLASSEN.

beatrubi commented:
Looks good from my point of view. Please let me know if I should trigger a merge.

zwif reviewed (approved):
Looks fine to me

paraenggu committed:
Adding base4kidsSchoolCycle attribute

The base4kidsSchoolCycle attribute defines to which cycle a given school class
belongs to.
It accepts the following values (accroding to Scolaris):
* Basisstufe (Zyklus 1): 1
* Kindergarten (Zyklus 1): 1
* Primarstufe (Zyklus 2): 2
* Sekundarstufe I (Zyklus 3): 3
* Sonderschulklassen (kein Zyklus): NULL (0 in the LDAP case)
